RecoMET/METAlgorithms/src/HcalHaloAlgo.cc

```diff
@@ -37,100 +36,92 @@ HcalHaloData HcalHaloAlgo::Calculate(const CaloGeometry& TheCaloGeometry, edm::H
 {
 
   HcalHaloData TheHcalHaloData;
-  
+  // ieta overlap geometrically w/ HB
+  const int iEtaOverlap = 22;
+  const int nPhiMax=73;
   // Store Energy sum of rechits as a function of iPhi (iPhi goes from 1 to 72)
-  float SumE[73];
+  float SumE[nPhiMax];
   // Store Number of rechits as a function of iPhi 
-  int NumHits[73];
+  int NumHits[nPhiMax];
   // Store minimum time of rechit as a function of iPhi
-  float MinTimeHits[73];
+  float MinTimeHits[nPhiMax];
   // Store maximum time of rechit as a function of iPhi
-  float MaxTimeHits[73];
-  for(unsigned int i = 0 ; i < 73 ; i++ ) 
+  float MaxTimeHits[nPhiMax];
+  for(unsigned int i = 0 ; i < nPhiMax ; i++ ) 
     {
       SumE[i] = 0;
       NumHits[i]= 0;
       MinTimeHits[i] = 0.;
       MaxTimeHits[i] = 0.;
     }
   
-  for( HBHERecHitCollection::const_iterator hit = TheHBHERecHits->begin() ; hit != TheHBHERecHits->end() ; hit++ )
-    {
-      HcalDetId id = HcalDetId(hit->id());                                                                                                    
-      switch ( id.subdet() )                                                                                         
-	{      
-	case HcalBarrel:                                                                           
-	  if(hit->energy() < HBRecHitEnergyThreshold )continue;
-	  break;                                                                                                                  
-	case HcalEndcap:                                                                                          
-	  if(hit->energy() < HERecHitEnergyThreshold ) continue;
-	  break;
-	default:
-	  continue;
-	}
-      
-      int iEta = id.ieta();
-      int iPhi = id.iphi();
-      if(iPhi < 73 && TMath::Abs(iEta) < 23 )
-	{ 
-	  SumE[iPhi]+= hit->energy();
-	  NumHits[iPhi] ++;
+  for (const auto & hit : (HBHERecHitCollection)(*TheHBHERecHits)) {
```

Comment:
hi @bsunanda - this cast should not be there. (and similarly below) - it likely causes a copy or worse ill defined behavior